<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>977</width>
    <height>799</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>cfg-editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="icon_layout">
      <item>
       <widget class="QToolButton" name="new_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>create new SEC node</string>
        </property>
        <property name="text">
         <string>New</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/new.png</normaloff>
          <normalon>:/cfg_editor/new.png</normalon>:/cfg_editor/new.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="open_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>open file</string>
        </property>
        <property name="text">
         <string>Open</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/open.png</normaloff>
          <normalon>:/cfg_editor/open.png</normalon>:/cfg_editor/open.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="save_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>save</string>
        </property>
        <property name="text">
         <string>Save</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/save.png</normaloff>
          <normalon>:/cfg_editor/save.png</normalon>:/cfg_editor/save.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="add_interface_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>add interface</string>
        </property>
        <property name="text">
         <string>Add interface</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/add_interface.png</normaloff>
          <normalon>:/cfg_editor/add_interface.png</normalon>:/cfg_editor/add_interface.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="add_module_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>add module</string>
        </property>
        <property name="text">
         <string>Add module</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/add_module.png</normaloff>
          <normalon>:/cfg_editor/add_module.png</normalon>:/cfg_editor/add_module.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="add_parameter_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>add parameter</string>
        </property>
        <property name="text">
         <string>Add parameter</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/add_parameter.png</normaloff>
          <normalon>:/cfg_editor/add_parameter.png</normalon>:/cfg_editor/add_parameter.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="add_property_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>add property</string>
        </property>
        <property name="text">
         <string>Add property</string>
        </property>
        <property name="icon">
         <iconset>
          <normalon>:/cfg_editor/add_property.png</normalon>
         </iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="add_comment_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>add comment</string>
        </property>
        <property name="text">
         <string>Add comment</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/add_comment.png</normaloff>:/cfg_editor/add_comment.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="lineWidth">
         <number>1</number>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="duplicate_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>duplicate module or interface</string>
        </property>
        <property name="text">
         <string>Duplicate</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/duplicate.png</normaloff>
          <normalon>:/cfg_editor/duplicate.png</normalon>:/cfg_editor/duplicate.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="delete_btn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>delete</string>
        </property>
        <property name="text">
         <string>Delete</string>
        </property>
        <property name="icon">
         <iconset resource="../../../../resources/frappy-gui.qrc">
          <normaloff>:/cfg_editor/delete.png</normaloff>
          <normalon>:/cfg_editor/delete.png</normalon>:/cfg_editor/delete.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>60</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextUnderIcon</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="main_layout">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="contextMenuPolicy">
         <enum>Qt::DefaultContextMenu</enum>
        </property>
        <property name="tabShape">
         <enum>QTabWidget::Rounded</enum>
        </property>
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="documentMode">
         <bool>false</bool>
        </property>
        <property name="tabsClosable">
         <bool>false</bool>
        </property>
        <property name="movable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>977</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="action_Close"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionAdd_interface"/>
    <addaction name="actionAdd_module"/>
    <addaction name="actionAdd_parameter"/>
    <addaction name="actionAdd_property"/>
    <addaction name="actionAdd_comment"/>
    <addaction name="actionDuplicate"/>
    <addaction name="actionDelete"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menu_Edit"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../../../../resources/frappy-gui.qrc">
     <normaloff>:/cfg_editor/new.png</normaloff>
     <normalon>:/cfg_editor/new.png</normalon>:/cfg_editor/new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>Create new file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../../../../resources/frappy-gui.qrc">
     <normaloff>:/cfg_editor/open.png</normaloff>
     <normalon>:/cfg_editor/open.png</normalon>:/cfg_editor/open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../../../../resources/frappy-gui.qrc">
     <normaloff>:/cfg_editor/save.png</normaloff>
     <normalon>:/cfg_editor/save.png</normalon>:/cfg_editor/save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save &amp;as ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAdd_SEC_node">
   <property name="text">
    <string>add SEC node</string>
   </property>
  </action>
  <action name="actionAdd_interface">
   <property name="text">
    <string>add interface</string>
   </property>
  </action>
  <action name="actionAdd_module">
   <property name="text">
    <string>add module</string>
   </property>
  </action>
  <action name="actionAdd_parameter">
   <property name="text">
    <string>add parameter</string>
   </property>
  </action>
  <action name="actionAdd_property">
   <property name="text">
    <string>add property</string>
   </property>
  </action>
  <action name="actionAdd_comment">
   <property name="text">
    <string>add comment</string>
   </property>
  </action>
  <action name="actionDuplicate">
   <property name="text">
    <string>duplicate</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>delete</string>
   </property>
  </action>
  <action name="action_Close">
   <property name="text">
    <string>&amp;Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F4</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../../../../resources/frappy-gui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>new_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>39</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>91</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>open_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>132</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>save_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>222</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_interface_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_interface()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>346</x>
     <y>83</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_module_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_module()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>432</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_parameter_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_parameter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_property_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_property()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>644</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>613</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>add_comment_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_comment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>732</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>708</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>duplicate_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_duplicate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>831</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>818</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>delete_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_delete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>919</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>909</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_as</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionSave_as()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionQuit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_actionAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_interface</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_interface()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_module</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_module()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_parameter</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_parameter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_property</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_property()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAdd_comment</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_add_comment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDuplicate</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_duplicate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_delete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Close</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_action_Close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_actionNew()</slot>
  <slot>on_add_module()</slot>
  <slot>on_add_interface()</slot>
  <slot>on_add_parameter()</slot>
  <slot>on_add_property()</slot>
  <slot>on_add_comment()</slot>
  <slot>on_actionOpen()</slot>
  <slot>on_actionSave()</slot>
  <slot>on_duplicate()</slot>
  <slot>on_delete()</slot>
  <slot>on_actionSave_as()</slot>
  <slot>on_actionQuit()</slot>
  <slot>on_actionAbout()</slot>
  <slot>on_action_Close()</slot>
 </slots>
</ui>
